databricks-sdk-java/src/main/java/com/databricks/sdk/core/commons/CommonsHttpClient.java

```diff
@@ -57,6 +76,99 @@ private CloseableHttpClient makeClosableHttpClient() {
         .build();
   }
 
+  private CloseableHttpClient makeClosableHttpClient(ProxyConfig proxyConfig) {
+    HttpClientBuilder builder =
+        HttpClientBuilder.create()
+            .setConnectionManager(connectionManager)
+            .setDefaultRequestConfig(makeRequestConfig());
+    setupProxy(proxyConfig, builder);
+    return builder.build();
+  }
+
+  public static void setupProxy(ProxyConfig config, HttpClientBuilder builder) {
+    String proxyHost = null;
+    Integer proxyPort = null;
+    String proxyUser = null;
+    String proxyPassword = null;
+    if (config.isUseSystemProperties()) {
+      builder.useSystemProperties();
+      String protocol = System.getProperty("https.proxyHost") != null ? "https" : "http";
+      proxyHost = System.getProperty(protocol + ".proxyHost");
+      proxyPort = Integer.parseInt(System.getProperty(protocol + ".proxyPort"));
+      proxyUser = System.getProperty(protocol + ".proxyUser");
+      proxyPassword = System.getProperty(protocol + ".proxyPassword");
+    }
+    // Override system properties if proxy configuration is explicitly set
+    if (config.getHost() != null) {
+      proxyHost = config.getHost();
+      proxyPort = config.getPort();
+      proxyUser = config.getUsername();
+      proxyPassword = config.getPassword();
+      builder.setProxy(new HttpHost(proxyHost, proxyPort));
+    }
+    setupProxyAuth(
+        proxyHost, proxyPort, config.getProxyAuthType(), proxyUser, proxyPassword, builder);
+  }
+
+  public static void setupProxyAuth(
+      String proxyHost,
+      Integer proxyPort,
+      ProxyConfig.ProxyAuthType proxyAuthType,
+      String proxyUser,
+      String proxyPassword,
+      HttpClientBuilder builder) {
+    AuthScope authScope = new AuthScope(proxyHost, proxyPort);
+    switch (proxyAuthType) {
+      case NONE:
+        break;
+      case BASIC:
+        setupBasicProxyAuth(builder, authScope, proxyUser, proxyPassword);
+        break;
+      case SPNEGO:
+        setupNegotiateProxyAuth(builder, authScope);
+        break;
+      default:
+        throw new DatabricksException("Unknown proxy auth type: " + proxyAuthType);
+    }
+  }
+
+  public static void setupNegotiateProxyAuth(HttpClientBuilder builder, AuthScope authScope) {
+    // We only support kerberos for negotiate as of now
+    System.setProperty("javax.security.auth.useSubjectCredsOnly", "false");
+    // "java.security.krb5.conf" system property needs to be set if krb5.conf is not in the default
+    // location
+    // Use "sun.security.krb5.debug" and "sun.security.jgss.debug" system properties for debugging
+    Credentials use_jaas_creds =
+        new Credentials() {
+          public String getPassword() {
+            return null;
+          }
+
+          public Principal getUserPrincipal() {
+            return null;
+          }
+        };
+
+    CredentialsProvider credsProvider = new BasicCredentialsProvider();
+    credsProvider.setCredentials(authScope, use_jaas_creds);
```

Comment:
but we are only returning null, do we still need to set this credential provider?

Reply:
Drawing an analogy from curl: https://curl.se/docs/manpage.html#-u
>If you use a Windows SSPI-enabled curl binary and perform Kerberos V5, Negotiate, NTLM or Digest authentication then you can tell curl to select the username and password from your environment by specifying a single colon with this option: "-u :".
I believe when both username and password are not provided is when it looks at the environment to pick up the credentials